Python/Product/Profiling/Profiling/UserInputService.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+namespace Microsoft.PythonTools.Profiling {
+    using System;
+    using System.ComponentModel.Composition;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Windows;
+    using Microsoft.PythonTools.Infrastructure;
+    using Microsoft.PythonTools.Interpreter;
+
+    /// <summary>
+    /// Implements a service to collect user input for profiling and generate a <see cref="TargetCommand"/>.
+    /// </summary>
+    [Export(typeof(IUserInputService))]
```

Comment:
The name of this seems weird to me. It feels like calling it IUserInputService is a bit of a misnomer. Like it should mention profiling somehow. It's only for profiling.

`IPythonProfilingCommandBuilder` maybe?

Reply:
But there's an actual `CommandBuilder` that converts user input to command arguments. How about I call the main service `PythonProfilingCommandArgumentProvider`, which calls `ArgumentInputDialog` to pop dialog to user and return user input, and calls `ArgumentParser` to convert user input to command arguments?

Reply:
The important part for me is the name indicates:
- What it's used for = in this case profiling
- What it does/returns = some sort of command? Or is it a target profile?